tests/test_memcache.py

```diff
@@ -58,6 +59,34 @@ def test_delete(self):
         self.assertEqual(result, True)
         self.assertEqual(self.mc.get("long"), None)
 
+        # Delete with explicit time=0 (can re-set immediately)
+        self.mc.set("my_key", "my_val")
+        self.mc.delete("my_key", time=0)
+        self.assertNotEqual(self.mc.set("my_key", "my_val"), 0)
```

Comment:
I added this test because that case (delete with time=0) wasn't covered by the test suite before, and there was a risk my `touch()` fix would change this behaviour. So I considered it as a safety net, with the added benefit of increasing test coverage.

`assertNotEqual(self.mc.set("my_key", "my_val"), 0)` to be consistent with the docstrings in `memcache.py`. It is stated:
- if `delete(time=0)`, subsequent `set()` should succeed immediately.
- a successful `set()` returns nonzero.